addons/mass_mailing/wizard/mailing_list_add_contact.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class MailingListAddContact(models.TransientModel):
+    _name = "mailing.list.add.contact"
+    _description = "Add contact to mailing list wizard"
+
+    contact_ids = fields.Many2many('mailing.contact')
+    mailing_list_id = fields.Many2one('mailing.list', string='Mailing List', required=True)
+
+    def action_add_contact_to_mailing_list(self):
+        self.ensure_one()
+        ctx = dict(self.env.context)
+        ctx.pop('default_state', None)
+        self = self.with_context(ctx)
+
+        for contact_id in self.contact_ids:
+            self.mailing_list_id.write({
+                'contact_ids': [(4, contact_id.id, 0)]
+            })
+
+        return {
+            'type': 'ir.actions.act_window',
+            'view_mode': 'form',
+            'res_model': 'mailing.list',
+            'target': 'current',
+            'res_id': self.mailing_list_id.id
+        }
```

Comment:
We try to avoid creating actions from scratch where there are actions that are close to what we want.
So here for example you would do:

```py
action = self.env["ir.actions.actions"]._for_xml_id("mass_mailing.action_view_mass_mailing_lists")
action['view_mode'] = 'form'
action['res_id'] = self.mailing_list_id.id
return action
```

That way if we define some specific behavior in the global action, it will also be applied here.